pkg/csi_driver/controller.go

```diff
@@ -81,7 +85,16 @@ func (s *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolu
 	if err := s.config.driver.validateVolumeCapabilities(req.GetVolumeCapabilities()); err != nil {
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
+	s.config.mutex.Lock()
+	instances, err := s.config.fileService.ListInstances(ctx, "-")
```

Comment:
We should avoid holding locks across calls that could potentially block for a long time.  I'm not sure if we need to lock in this case since each CreateVolume is doing its own ListInstances call

Reply:
I would also move this call to just before `CreateInstance` to avoid unnecessary cloud provider calls

Reply:
To read the updated list of reserved IPs, we have to wait until this call succeeds right?

Reply:
I think you are thinking about the scenario on how to coordinate multiple CreateVolume requests and make them select different IPs.  

One way to do it is to lock the entire CreateVolume call so only one volume create can occur at a time. This will work, but I think it will be undesirable because every cloud call could take on the order of minutes.  We could try this way just for now to get it working, but we'll definitely need to improve on it.

In the end, we want to allow parallel CreateVolume operations to occur, but need to serialize IP allocation.  I think this will require some sort of caching/shared allocator.